src/test/java/com/mossle/cdn/CdnUtilsTest.java

```diff
@@ -0,0 +1,100 @@
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @PrepareForTest({CdnUtils.class})
+    @Test
+    public void testCopyUrlToFile() throws Exception {
+        PowerMockito.mockStatic(UUID.class);
+        Date date = new Date(1560600794000L);
+        PowerMockito.whenNew(Date.class).withNoArguments().thenReturn(date);
+        UUID uUID = PowerMockito.mock(UUID.class);
+        PowerMockito.when(uUID.randomUUID().toString()).thenReturn("baz");
+
+        Assert.assertEquals("/2019/06/15/baz.uk/",
+                CdnUtils.copyUrlToFile(
+                        "foo", "https://www.bbc.co.uk/", "baz"));
+        Assert.assertEquals("/2019/06/15/baz.uk/",
+                CdnUtils.copyUrlToFile(
+                        "foo", "https://www.bbc.co.uk/", "baz", null));
+
+        thrown.expect(IllegalStateException.class);
+        CdnUtils.copyUrlToFile("foo", "https://www.bbc.co.uk/", "baz", "../");
+    }
+
+    @PrepareForTest({CdnUtils.class})
+    @Test
+    public void testCopyMultipartFileToFile() throws Exception {
+        PowerMockito.mockStatic(UUID.class);
+        Date date = new Date(1560600794000L);
+        PowerMockito.whenNew(Date.class).withNoArguments().thenReturn(date);
+        UUID uUID = PowerMockito.mock(UUID.class);
+        PowerMockito.when(uUID.randomUUID().toString()).thenReturn("baz");
+
+        MultipartFile mFile = new MockMultipartFile("foo", new byte[0]);
+        Assert.assertEquals("/2019/06/15/baz",
+                CdnUtils.copyMultipartFileToFile(
+                        "foo", mFile, "bar", null));
+
+        thrown.expect(IllegalStateException.class);
+        CdnUtils.copyMultipartFileToFile("foo", mFile, "bar", "../");
+    }
+
+    @PrepareForTest({CdnUtils.class})
+    @Test
+    public void testGenerateTargetFileName() throws Exception {
+        PowerMockito.mockStatic(UUID.class);
+        Date date = new Date(1560600794000L);
+        PowerMockito.whenNew(Date.class).withNoArguments().thenReturn(date);
+        UUID uUID = PowerMockito.mock(UUID.class);
+        PowerMockito.when(uUID.randomUUID().toString()).thenReturn("baz");
+
+        Assert.assertEquals("/2019/06/15/baz",
+                CdnUtils.generateTargetFileName("foo"));
+    }
+
+    @PrepareForTest({CdnUtils.class})
+    @Test
+    public void testFindTargetFile() throws IOException {
+        File file = folder.newFile("foo.txt");
+
+        Assert.assertEquals(file,
+                CdnUtils.findTargetFile(file.getPath(), "", ""));
+        Assert.assertEquals(new File("foo/bar"),
+                CdnUtils.findTargetFile(
+                        "foo", "bar", file.getName()).getParentFile());
+    }
+
+    @Test
+    public void testFindSuffix() {
+        Assert.assertEquals("", CdnUtils.findSuffix("1a 2b 3c"));
+        Assert.assertEquals(".uk/",
+                CdnUtils.findSuffix("https://www.bbc.co.uk/?bar"));
```

Comment:
Was the BBC website something test-gen came up with or something you put in? If it's the latter can it be something like `https://www.foo.bar/?baz` or something?

Reply:
Same for the other occurrences of it in the tests.

Reply:
^ agree, maybe just `example.com`

Reply:
changed to `"https://www.foo.bar/?baz"`